examples/tracing-grpc/src/client.rs

```diff
@@ -24,45 +39,46 @@ pub mod hello_world {
     tonic::include_proto!("helloworld");
 }
 
-#[instrument]
 async fn greet() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
-    let mut client = GreeterClient::connect("http://[::1]:50051")
-        .instrument(info_span!("client connect"))
-        .await?;
+    let tracer = global::tracer("example/client");
+    let span = tracer
+        .span_builder(String::from("Greeter/client"))
+        .with_kind(SpanKind::Client)
+        .with_attributes(vec![KeyValue::new("component", "grpc")])
+        .start(&tracer);
+    let cx = Context::current_with_span(span);
+    let mut client = GreeterClient::connect("http://[::1]:50051").await?;
 
     let mut request = tonic::Request::new(HelloRequest {
         name: "Tonic".into(),
     });
 
     global::get_text_map_propagator(|propagator| {
-        propagator.inject_context(
-            &tracing::Span::current().context(),
-            &mut MetadataMap(request.metadata_mut()),
-        )
+        propagator.inject_context(&cx, &mut MetadataMap(request.metadata_mut()))
     });
 
-    let response = client
-        .say_hello(request)
-        .instrument(info_span!("say_hello"))
-        .await?;
+    let response = client.say_hello(request).await;
+
+    let status = match response {
+        Ok(_res) => "OK".to_string(),
+        Err(status) => {
+            // Access the status code
+            let status_code = status.code();
+            status_code.to_string()
+        }
+    };
+    cx.span().add_event(
+        "Got response!".to_string(),
+        vec![KeyValue::new("status", status)],
```

Comment:
I think we should be showing the recommended approach of storing this to Span status itself, if failed.